ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/HostInfoEntity.java

```diff
@@ -26,15 +27,22 @@
   private final String host;
   private final int port;
 
-  @JsonCreator
   public HostInfoEntity(
-      @JsonProperty("host") final String host,
-      @JsonProperty("port") final int port
+      final String host,
+      final int port
   ) {
     this.host = Objects.requireNonNull(host, "host");
     this.port = Objects.requireNonNull(port, "port");
   }
 
+  @JsonCreator
+  public HostInfoEntity(final String serializedPair) {
+    final String [] parts = serializedPair.split(":");
+    Preconditions.checkArgument(parts.length == 2);
+    this.host = Objects.requireNonNull(parts[0], "host");
+    this.port = Integer.parseInt(parts[1]);
+  }
```

Comment:
This will throw a number parsing error if the second part is not a number, which ain't nice.

How about:

```java
Can we throw a more descriptive exception here? e.g. 

```java
final String [] parts = serializedPair.split(":");
if (parts.length != 2) {
   throw new KsqlException("Invalid host info. Expected format: <hostname>:<port>, but was " + seirlizedPair;
}

this.host = Objects.requireNonNull(parts[0], "host");

try {
   this.port = Integer.parseInt(parts[1]);
} catch (final Exception e) {
   throw new KsqlException("Invalid host info. Expected format: <hostname>:<port>, but was " + seirlizedPair, e);
}
```